website/docs/reference/global-configs.md

```diff
@@ -103,46 +103,55 @@ config:
 
 </File>
 
-## Failing fast
+<VersionBlock firstVersion="1.1">
 
-Supply the `-x` or `--fail-fast` flag to `dbt run` to make dbt exit immediately if a single resource fails to build. If other models are in-progress when the first model fails, then dbt will terminate the connections for these still-running models. 
+### Cache database objects for selected resource
 
-For example, you can select four models to run, but if a failure occurs in the first model, the failure will prevent other models from running:
+:::caution Experimental config flag
+This should be considered an _unstable_ config flag because it is experimental and subject to change. We reserve the right to make breaking changes to this config flag.
+:::
+
+At the start of runs, dbt caches metadata about all the objects in all the schemas where it might materialize resources (such as models). By default, dbt caches all schemas related to the project. When this config is enabled, dbt will only cache schemas related to selected resources for the current run. This can offer significant speed improvements when running a small subset of a large project.
+
+For example, to improve speed and performance while focused on Salesforce models you would select the source and use the `cache-selected-only` flag:
 
 ```text
-$ dbt -x run --threads 1
-Running with dbt=1.0.0
-Found 4 models, 1 test, 1 snapshot, 2 analyses, 143 macros, 0 operations, 1 seed file, 0 sources
 
-14:47:39 | Concurrency: 1 threads (target='dev')
-14:47:39 |
-14:47:39 | 1 of 4 START table model test_schema.model_1........... [RUN]
-14:47:40 | 1 of 4 ERROR creating table model test_schema.model_1.. [ERROR in 0.06s]
-14:47:40 | 2 of 4 START view model test_schema.model_2............ [RUN]
-14:47:40 | CANCEL query model.debug.model_2....................... [CANCEL]
-14:47:40 | 2 of 4 ERROR creating view model test_schema.model_2... [ERROR in 0.05s]
+$ dbt build --select source:salesforce 
+$ dbt --cache-selected-only run
```

Comment:
Let's combine these into a single command:
```suggestion
$ dbt --cache-selected-only run --select salesforce
```
In order to work, that single command must include the config and the selection criteria, and it must be selecting buildable resources (models, not sources).